x-pack/plugins/observability_ai_assistant/server/service/chat_function_client/index.ts

```diff
@@ -6,29 +6,85 @@
  */
 /* eslint-disable max-classes-per-file*/
 
-import type { ErrorObject, ValidateFunction } from 'ajv';
-import { keyBy } from 'lodash';
-import type {
+import Ajv, { type ErrorObject, type ValidateFunction } from 'ajv';
+import dedent from 'dedent';
+import { compact, keyBy } from 'lodash';
+import {
   ContextDefinition,
   ContextRegistry,
   FunctionResponse,
+  FunctionVisibility,
   Message,
+  ObservabilityAIAssistantAppContext,
+  RegisterContextDefinition,
 } from '../../../common/types';
 import { filterFunctionDefinitions } from '../../../common/utils/filter_function_definitions';
-import type { FunctionHandler, FunctionHandlerRegistry } from '../types';
+import type { FunctionHandler, FunctionHandlerRegistry, RegisterFunction } from '../types';
 
 export class FunctionArgsValidationError extends Error {
   constructor(public readonly errors: ErrorObject[]) {
     super('Function arguments are invalid');
   }
 }
 
+const ajv = new Ajv({
+  strict: false,
+});
+
 export class ChatFunctionClient {
-  constructor(
-    private readonly contextRegistry: ContextRegistry,
-    private readonly functionRegistry: FunctionHandlerRegistry,
-    private readonly validators: Map<string, ValidateFunction>
-  ) {}
+  private readonly contextRegistry: ContextRegistry = new Map();
+  private readonly functionRegistry: FunctionHandlerRegistry = new Map();
+  private readonly validators: Map<string, ValidateFunction> = new Map();
+
+  constructor(private readonly appContexts: ObservabilityAIAssistantAppContext[]) {
+    const allData = compact(appContexts.flatMap((context) => context.data));
+
+    if (allData.length) {
+      this.registerFunction(
+        {
+          name: 'get_data_on_screen',
```

Comment:
I want to make sure I understand this correctly: The LLM can now get the chat context in two ways: 
 - calling the `context` function will return the chat context (possibly truncated due to token limit) or
 - calling the `get_data_on_screen` function which will return the untruncated chat context

Reply:
Btw. it looks like `get_data_on_screen`  doesn't include the `appContexts.description` (aka "screenDescription"). Is that intentional? Afaict the context for the SLOList only has screen description and no `data` attribute so that would result in no context being provided here

Reply:
Yes, the descriptions are automatically added in the `context` function call, so there is no need for the LLM to request it.